src/Symfony/Component/Notifier/Bridge/MsTeams/MsTeamsTransportFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\MsTeams;
+
+use Symfony\Component\Notifier\Exception\IncompleteDsnException;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Edouard Lescot <edouard.lescot@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class MsTeamsTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+        $host = $dsn->getHost();
+        $port = $dsn->getPort();
+
+        if ('teams' === $scheme) {
+            return (new MsTeamsTransport($this->getPath($dsn), $this->client, $this->dispatcher))->setHost($host)->setPort($port);
+        }
+
+        throw new UnsupportedSchemeException($dsn, 'teams', $this->getSupportedSchemes());
```

Comment:
```suggestion
        $scheme = $dsn->getScheme();
        
        if ('teams' !== $scheme) {
            throw new UnsupportedSchemeException($dsn, 'teams', $this->getSupportedSchemes());
        }
        
        $path = $dsn->getPath();
        if (null === $path) {
            throw new IncompleteDsnException('Path is not set.', $dsn->getOriginalDsn());
        }
        
        $host = $dsn->getHost();
        $port = $dsn->getPort();

        return (new MsTeamsTransport($this->getPath($dsn), $this->client, $this->dispatcher))->setHost($host)->setPort($port);
```

No need for an extra method